src/repository.c

```diff
@@ -946,28 +954,48 @@ bool git_repository__reserved_names(
 }
 #endif
 
-static int check_repositoryformatversion(git_config *config)
+static int check_repositoryformatversion(int *version, git_config *config)
 {
-	int version, error;
+	int error;
 
-	error = git_config_get_int32(&version, config, "core.repositoryformatversion");
+	error = git_config_get_int32(version, config, "core.repositoryformatversion");
 	/* git ignores this if the config variable isn't there */
-	if (error == GIT_ENOTFOUND)
-		return 0;
+	if (error == GIT_ENOTFOUND) {
+		giterr_clear();
+		return GIT_REPO_VERSION;
```

Comment:
We return our usual 0 or -1 to indicate success. Though `GIT_REPO_VERSION` is 0 here, we should still have a `return 0;` here.